ext/sockets/sockets.c

```diff
@@ -2586,14 +2586,35 @@ PHP_FUNCTION(socket_addrinfo_lookup)
 	if (zhints && !HT_IS_PACKED(Z_ARRVAL_P(zhints))) {
 		ZEND_HASH_MAP_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(zhints), key, hint) {
 			if (key) {
+				bool failed = false;
 				if (zend_string_equals_literal(key, "ai_flags")) {
-					hints.ai_flags = zval_get_long(hint);
+					zend_long val = zval_try_get_long(hint, &failed);
+					if (failed) {
+						zend_argument_type_error(3, "\"ai_flags\" key must be of type int, %s given", zend_zval_type_name(hint));
+						RETURN_THROWS();
+					}
+					hints.ai_flags = (int)val;
```

Comment:
Probably there should be a range check to see if the val can fit in a C int, but I'm happy to leave this for a follow-up PR